### PR TITLE
hardnested SIMD select

### PR DESCRIPTION
**Pull request by merlokk**
#404 

something like that.
its OK?
complete it?

### CODE REVIEW

File: client/cmdhfmfhard.c [hunk 1]
```diff
@@ -2528,6 +2534,8 @@ static void set_test_state(uint8_t byte)
 int mfnestedhard(uint8_t blockNo, uint8_t keyType, uint8_t *key, uint8_t trgBlockNo, uint8_t trgKeyType, uint8_t *trgkey, bool nonce_file_read, bool nonce_file_write, bool slow, int tests) 
 {
 	char progress_text[80];
+	
+	SetSIMDInstr(SIMD_NONE);
```
Comment (merlokk) [inline client/cmdhfmfhard.c]: this needs to set from commandline

**Comment by merlokk** [comment]
To look. If this way is acceptable - i will complete this PR
if not - just close it.

**Comment by iceman1001** [comment]
ok,  this puppy is @pwpiwi 's :) 

Btw, you are hard to get in touch with...   Tried emailing u.
Would you mind testing icemanfork and see if it suffers from same bug (it should but..well,  can't test it)

**Comment by merlokk** [comment]
my email: olegmsn@gmail.com, skype: olegmsn )))

**Comment by merlokk** [comment]
ok, now it works.

**Comment by pwpiwi** [comment]
> ok, now it works.

What works? hardnested with automatic instruction set detection?

**Comment by merlokk** [comment]
hardnested with auto and manual instruction set

but for my pc works only mmx and no SIMD. SSE2 not works and i dont know how to make it work.
but i can choose because code with auto detection instruction set just kills the process

**Comment by merlokk** [comment]
sory( wrong button....

**Comment by pwpiwi** [comment]
Hmmm. If we assume that the root cause analysis in #404 is correct, then we would need the same for the avx, avx2 and avx512. I am not sure about the performance impact - probably not too high because most of the SIMD functions will probably be inlined anyway - but why not simply add `__attribute__((force_align_arg_pointer))` here: https://github.com/Proxmark/proxmark3/blob/1c313691e6fbd76c48122c8c63844e240a6fb0d1/client/hardnested/hardnested_bruteforce.c#L143
This should cover all eventualities and has lowest impact.

**Comment by merlokk** [comment]
I thought that maybe it lays somewhere else too. but not) 2nd way works)

**Comment by merlokk** [comment]
done)

and......
I dont think that CPU from Intel and AMD may have such a big difference and
as for me Intel doing correctly because for AMD you need to have for address 4-bytes length all 16 bytes(

**Comment by merlokk** [comment]
#501

**Pull request merged**